src/mbgl/text/glyph_atlas.hpp

```diff
@@ -92,9 +104,19 @@ class GlyphAtlas : public util::noncopyable {
     };
 
     std::unordered_map<FontStack, Entry, FontStackHash> entries;
-    std::mutex mutex;
+    
+    struct TileDependency {
+        TileDependency(const GlyphRangeDependencies& _pendingRanges, GlyphDependencies _glyphDependencies, GlyphRequestor* _requestor)
+            : pendingRanges(_pendingRanges), glyphDependencies(std::move(_glyphDependencies)), requestor(_requestor)
+        {}
+        GlyphRangeDependencies pendingRanges;
+        GlyphDependencies glyphDependencies;
+        GlyphRequestor* requestor;
+    };
+    std::unordered_map<uintptr_t,TileDependency> tileDependencies;
+    typedef std::pair<FontStack,GlyphRange> PendingGlyphRange;
+    std::map<PendingGlyphRange, std::set<uintptr_t>> pendingGlyphRanges;
```

Comment:
Instead of adding a new `(FontStack, GlyphRange)`-keyed map, can we add a `std::set<GlyphRequestor*>` to `GlyphPBF` and perform a `FontStack` ⇢ `GlyphRange` ⇢ `GlyphPBF` lookup via `GlyphAtlas::entries` and `Entry::ranges`?